common/ferrostar/src/models.rs

```diff
@@ -59,18 +59,51 @@ impl From<Rect> for BoundingBox {
     }
 }
 
+/// The heading of the user/device.
+/// 
+/// Ferrostar prefers course over ground
+#[derive(Clone, Copy, PartialEq, PartialOrd, Debug, uniffi::Record)]
+pub struct Heading {
+    pub geographic_heading: f64,
+    pub accuracy: f64,
+    pub x: f64,
+    pub y: f64,
+    pub z: f64,
+    pub timestamp: SystemTime,
+}
+
+impl Heading {
+    pub fn new(
+        geographic_heading: f64,
```

Comment:
Convert to `true_heading`. Convert to int and add docstring.

Reply:
And switching to unsigned int avoids the negative invalid heading value.

Reply:
Add one or more unit tests to test a negative here to see if it gracefully fails when iOS gives us -1 "invalid"